core/chain.py

```
@@ -1,22 +1,92 @@
     """
 
-    def __init__(self):
+    def __init__(self, chain_file: Optional[str] = None):
         self.chain = []
         self.state = State()
         self._lock = threading.RLock()
-        self._create_genesis_block()
+        self._chain_file = chain_file or DEFAULT_CHAIN_FILE
+        self._load_from_file()
+
+    def _load_from_file(self):
```

Ruff (0.15.1)

[warning] Missing return type annotation for private function `_load_from_file`

Add return type annotation: `None`

(ANN202)